root/static/scripts/edit/components/forms.js

```diff
@@ -59,3 +59,34 @@ MB.initializeArtistCredit = function (form, initialArtistCredit) {
     );
   });
 };
+
+/*
+ * Registers a beforeunload event listener on the window that prompts
+ * the user if any of the page's form inputs have been changed.
+ */
+MB.installFormUnloadWarning = function () {
+  let modified = false;
+
+  const form = document.querySelector('#page form');
+
+  // This is somewhat heavy-handed, in that it will still warn even if the
+  // user changes an input back to its original value.
+  form.addEventListener('change', () => {
+    modified = true;
+  });
+
+  // Disarm the warning when the form is being submitted.
+  form.addEventListener('submit', () => {
+    modified = false;
+  });
+
+  window.addEventListener('beforeunload', event => {
+    if (!modified) {
+      return false;
```

Comment:
The release editor code returns true in the no-changes case, which seems odd to me: https://github.com/metabrainz/musicbrainz-server/blob/ef9817a85c7cae102aaf3ce8277ca78dbf48fe0a/root/static/scripts/release-editor/init.js#L260-L277

https://developer.mozilla.org/en-US/docs/Web/API/Window/beforeunload_event says that returning a truthy value will result in the prompt being shown, but only if the `onbeforeunload` property is used rather than `addEventListener`, which is probably why it's able to get away with it.

Reply:
Looks like it was added in https://github.com/metabrainz/musicbrainz-server/commit/6e3446d27e938e490cd381d154524cd21a95cd14 to satisfy an eslint rule, not for any technical reason.